core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -167,6 +168,8 @@
     private final IdentityHashMap<LikePredicate, JoniRegexp> likePatternCache = new IdentityHashMap<>();
     private final IdentityHashMap<InListExpression, Set<?>> inListCache = new IdentityHashMap<>();
 
+    private final HashBasedTable<OperatorType, List<? extends Type>, ResolvedFunction> operatorCache = HashBasedTable.create();
```

Comment:
It can grow unbounded, so this doesn't seem a safe change.

Reply:
ExpressionInterpreter is created per expression, its lifecycle is short.
The existed inListCache and likePatternCache are also unbounded, so I follow the style.

I also considered adding a bounded thread-safe cache in MetadataManager, but I am not sure if it is proper.